### PR TITLE
Replace runner gRPC by LCD in e2e tests

### PR DESCRIPTION
**Pull request by NicolasMahe**
Dependent on https://github.com/mesg-foundation/engine/pull/1751
Related to https://github.com/mesg-foundation/engine/issues/1741

- fix bug runner/builder/container when current UID and GID are not 0
- Introduce a new structure `runner/builderLCD` that create and stop docker services and returns the msg to broadcast.
- Use it in e2e tests.
- Remove gRPC RunnerClient from e2e test.

### CODE REVIEW

File: e2e/runner_test.go [hunk 1]
```diff
@@ -26,60 +28,38 @@ func testRunner(t *testing.T) {
 		require.NoError(t, err)
 		acknowledgement.WaitForStreamToBeReady(stream)
 
-		resp, err := client.RunnerClient.Create(context.Background(), &pb.CreateRunnerRequest{
-			ServiceHash: testServiceHash,
-			Env:         []string{"BAR=3", "REQUIRED=4"},
-		})
+		msg, err := runBuilder.Start(engineAddress, testServiceStruct, []string{"BAR=3", "REQUIRED=4"})
```
Comment (antho1404) [inline e2e/runner_test.go]: I would start the service and then when started, create the message directly in the test with the result of the start (if needed) this way the start is not related to a message just to its function of starting a runner.

File: e2e/runner_test.go [hunk 2]
```diff
@@ -26,60 +28,38 @@ func testRunner(t *testing.T) {
...
+		testRunnerHash = lcdBroadcastMsg(msg)
 
 		// wait for service to be ready
 		_, err = stream.Recv()
 		require.NoError(t, err)
 	})
 
 	t.Run("recreate", func(t *testing.T) {
-		_, err := client.RunnerClient.Delete(context.Background(), &pb.DeleteRunnerRequest{Hash: testRunnerHash})
+		msgDelete, err := runBuilder.Stop(engineAddress, testRunnerHash, testServiceStruct, true)
```
Comment (antho1404) [inline e2e/runner_test.go]: same here separating the stoping function to the actual message

File: runner/builder/builder_lcd.go [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+
+	inst, err := instance.New(srv.Hash, envHash)
+	if err != nil {
+		return runnermodule.MsgCreate{}, err
+	}
+	expRunner, err := runner.New(owner.String(), inst.Hash)
+	if err != nil {
+		return runnermodule.MsgCreate{}, err
+	}
+	expRunnerHash := expRunner.Hash
```
Comment (antho1404) [inline runner/builder/builder_lcd.go]: All this could be done by calling the querier `/runner/hash`

File: runner/builder/builder_lcd.go [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+	return &LCD{
+		container:    container,
+		engineName:   engineName,
+		port:         port,
+		ipfsEndpoint: ipfsEndpoint,
+	}
+}
+
+// Start start a new runner.
+func (b *LCD) Start(owner sdk.AccAddress, srv *service.Service, env []string) (runnermodule.MsgCreate, error) {
```
Comment (antho1404) [inline runner/builder/builder_lcd.go]: This builder package is not related to the LCD so it shouldn't return data for the LCD

File: e2e/runner_test.go (hunk 1 above)
Comment (antho1404) [reply e2e/runner_test.go]: You could even use the querier `/runner/hash` and you would have all the information to call the functions
- `build(cont container.Container, srv *service.Service, ipfsEndpoint string)`
- `start(cont container.Container, srv *service.Service, instanceHash hash.Hash, runnerHash hash.Hash, imageHash string, env []string, engineName, port string)`

That we can put public. This way the whole LCD struct in the builder package can be removed

File: runner/builder/builder_lcd.go (hunk 3 above)
Comment (NicolasMahe) [reply runner/builder/builder_lcd.go]: I thought the hash endpoints were just temporary?
But yeah, if so, it will shrink this function as much as nothing: the e2e could directly call the private functions of this package (that should be converted to public ones)

File: runner/builder/builder_lcd.go (hunk 4 above)
Comment (NicolasMahe) [reply runner/builder/builder_lcd.go]: with the other modification done, I will remove it

File: e2e/runner_test.go (hunk 1 above)
Comment (NicolasMahe) [reply e2e/runner_test.go]: done

File: e2e/runner_test.go (hunk 2 above)
Comment (NicolasMahe) [reply e2e/runner_test.go]: done

File: runner/builder/builder_lcd.go (hunk 4 above)
Comment (NicolasMahe) [reply runner/builder/builder_lcd.go]: done

File: runner/builder/builder_lcd.go (hunk 3 above)
Comment (NicolasMahe) [reply runner/builder/builder_lcd.go]: done
